src/elisp/treemacs-mouse-interface.el

```diff
@@ -300,7 +318,10 @@ and ignore any prefix argument."
             (menu
              (easy-menu-create-menu
               nil
-              `(("New"
+              `(["Paste here"
+                 treemacs--paste-point-to-minibuffer
+                 :visible ,(string-match-p "\\(\\(Move\\)\\|\\(Copy\\)\\) to: " (or (minibuffer-prompt) ""))]
+                ("New"
```

Comment:
Finally at the end the commits need to be squashed and the commit message prefixed with `[Interface]`.